consensus/src/commit_notifier.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::QuorumStoreError;
+use anyhow::{format_err, Result};
+use aptos_infallible::Mutex;
+use aptos_metrics::monitor;
+use consensus_types::{common::Round, request_response::ConsensusRequest};
+use futures::channel::{mpsc, mpsc::Sender, oneshot};
+use std::time::Duration;
+use tokio::time::timeout;
+
+/// Notification of consensus committed logical time for QuorumStore to clean.
+#[async_trait::async_trait]
+pub trait CommitNotifier: Send + Sync {
+    /// Notification of committed logical time
+    async fn notify_commit(&self, epoch: u64, round: Round) -> Result<(), QuorumStoreError>;
+
+    fn new_epoch(&self, quorum_store_commit_sender: mpsc::Sender<ConsensusRequest>);
+}
```

Comment:
yep, basically whenever we know quorum has enough info persisted about {epoch, round} that it would be possible to state-sync, we should let QS component (currently batchreader) know.